pkg/geodata/geodata.go

```diff
@@ -150,6 +153,77 @@ AND nomis_category.year = %d
 	return app.collectCells(ctx, sql)
 }
 
+// radiusQuery returns the csv table for areas within radius meters from location
+//
+func (app *Geodata) radiusQuery(ctx context.Context, location string, radius int, geotype string, cats []string) (string, error) {
+	var lon, lat float64
+	fields, err := fmt.Sscanf(location, "%f,%f", &lon, &lat)
+	if err != nil {
+		return "", fmt.Errorf("scanning location %q: %w", location, err)
+	}
+	if fields != 2 {
+		return "", fmt.Errorf("location missing a number: %w", ErrMissingParams)
+	}
+
+	if radius < 1 {
+		return "", fmt.Errorf("radius must be >0: %q: %w", radius, err)
+	}
+
+	if geotype == "" {
+		return "", fmt.Errorf("geotype required: %w", ErrMissingParams)
+	}
+
+	template := `
+SELECT
+    geo.code AS geography_code,
+    nomis_category.long_nomis_code AS category_code,
+    geo_metric.metric AS value
+FROM
+    geo,
+    geo_type,
+    geo_metric,
+    data_ver,
+    nomis_category
+WHERE ST_DWithin(
+    geo.wkb_geometry::geography,
+    ST_SetSRID(
+        ST_Point(%f, %f),
+        4326
+    )::geography,
+    %d
```

Comment:
As this is geography, are the units for the radius meters? https://postgis.net/docs/ST_DWithin.html

Reply:
ah yeah just re-read the help text above properly - yes they are